### PR TITLE
sql-tasks. Dzmitry Bryt

### PR DESCRIPTION
**Pull request by Daqueze21**
https://travis-ci.org/github/Daqueze21/db-assignments/builds/723433353

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -91,7 +129,18 @@ async function task_1_5(db) {
  *
  */
 async function task_1_6(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT 
+		ProductName,
+		Categories.CategoryName,
+		Suppliers.CompanyName AS 'SupplierCompanyName'
+            FROM ((Products
```
Comment (AlexMokin) [inline task/sql-tasks.js]: remove redundant parentheses

File: task/sql-tasks.js [hunk 2]
```diff
@@ -177,7 +279,13 @@ async function task_1_12(db) {
  *
  */
 async function task_1_13(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT 
+            count(*) AS TotalOfCurrentProducts,
+            SUM(Discontinued) AS TotalOfDiscontinuedProducts
```
Comment (AlexMokin) [inline task/sql-tasks.js:285]: use count when you need to count recrods to show intent explicitly

Comment (AlexMokin) [reply task/sql-tasks.js:285]: but even better to write this query like this

```
select
(select count()...) as 'T'
(select count()...) as 'TT'
```

notice that from-clause is optional

File: task/sql-tasks.js [hunk 3]
```diff
@@ -199,7 +315,24 @@ async function task_1_14(db) {
  *
  */
 async function task_1_15(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT 
+            SUM(MONTH(OrderDate)=1) AS 'January',
+            SUM(MONTH(OrderDate)=2) AS 'February',
+            SUM(MONTH(OrderDate)=3) AS 'March',
+            SUM(MONTH(OrderDate)=4) AS 'April',
```
Comment (AlexMokin) [inline task/sql-tasks.js]: use count when you need to count recrods to show intent explicitly

File: task/sql-tasks.js [hunk 4]
```diff
@@ -280,7 +471,25 @@ async function task_1_21(db) {
...
+        SELECT DISTINCT 
+        Customers.CompanyName,
+		Products.ProductName,
+		OrderDetails.UnitPrice as 'PricePerItem'
+        FROM Customers
+        LEFT JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+        LEFT JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+        LEFT JOIN Products ON OrderDetails.ProductID = Products.ProductID
+        WHERE OrderDetails.UnitPrice = (SELECT MAX(odt.UnitPrice) as 'MaxPrice'     FROM Customers cus
+            INNER JOIN Orders ord ON Customers.CustomerID = ord.CustomerID
```
Comment (AlexMokin) [inline task/sql-tasks.js]: try to group by customer id and join this subquery on unitprice and customerid

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IlyaMokin** [comment]
The task is accepted! Thank you for the efforts!

**Pull request closed without merging**